kythe/cxx/common/kzip_writer_c_api.h

```diff
@@ -0,0 +1,98 @@
+#include <stddef.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+#define KZIP_WRITER_ENCODING_JSON (1)
+#define KZIP_WRITER_ENCODING_PROTO (2)
+#define KZIP_WRITER_ENCODING_ALL \
+  (KZIP_WRITER_ENCODING_JSON | KZIP_WRITER_ENCODING_PROTO)
+
+#define KZIP_WRITER_BUFFER_TOO_SMALL_ERROR (-1)
+#define KZIP_WRITER_PROTO_PARSING_ERROR (-2)
+
+/// \brief The opaque kzip writer object.
+struct KzipWriter;
+
+/// \brief Creates a new kzip writer.
+/// \param path The path to the file to create.
+/// \param path_len the string length of path, required due to the C API.
+/// \param encoding The compilation unit encoding, either
+///        KZIP_WRITER_ENCODING_JSON, or KZIP_WRITER_ENCODING_PROTO
+/// \param create_status zero if created fine, nonzero in case of an error.
+///        Positive error values are the same as in absl::Status.  Negative
+///        error values are either KZIP_WRITER_BUFFER_TOO_SMALL_ERROR, or
+///        KZIP_WRITER_PROTO_PARSING_ERROR.
+/// Caller takes the ownership of the returned pointer.  KzipWriter_Close must
+/// be called to release the memory.
+struct KzipWriter* KzipWriter_Create(const char* path, const size_t path_len,
+                                     int32_t encoding, int32_t* create_status);
+
+/// \brief Deallocates KzipWriter.
+///
+/// Must be called once KzipWriter is no longer needed.
+void KzipWriter_Delete(struct KzipWriter* writer);
+
+/// \brief Closes the writer.
+/// Must be called before destroying the object.
+int32_t KzipWriter_Close(struct KzipWriter* writer);
+
+/// \brief Writes a piece of content into the kzip, returning a digest.
+///
+/// The content is
+/// specified using both the pointer to the beginning and the size.  The caller
+/// must provide a buffer to put the digest into, and a buffer size.  Nonzero
+/// status is returned in case of an error writing.  Specifically, if there is
+/// insufficient space in the buffer, KZIP_WRITER_BUFFER_TOO_SMALL_ERROR is
+/// returned.
+/// \param writer The writer to write into.
+/// \param content The file content buffer to write.
+/// \param content_length The length of the buffer to write.
+/// \param digest_buffer The output buffer to store the digest into.  Must be
+///        large enough to hold a digest.
+/// \param buffer_length The length of digest_buffer in bytes.
+int32_t KzipWriter_WriteFile(struct KzipWriter* writer, const char* content,
+                             const size_t content_length, char* digest_buffer,
+                             size_t buffer_length,
+                             size_t* resulting_digest_size);
+
+/// \brief Writes a compilation unit into the kzip, returning a digest.
+//
+/// The caller must provide a buffer to put the digest into, and a buffer size.
+/// Nonzero status is returned in case of an error while writing.
+int32_t KzipWriter_WriteUnit(struct KzipWriter* writer, const char* proto,
+                             size_t proto_length, char* digest_buffer,
```

Comment:
Document the lack of null-termination here (and in the other output buffers)

Reply:
Done.